File: Problem Solving/Algorithms/Implementation/Number Line Jumps.c
char* kangaroo(int x1, int v1, int x2, int v2)
{
    while (x1 < x2)
    {
        x1 += v1;
        x2 += v2;
    }
    
    if (x1 == x2)
    {
        return "YES";
    }
    else
    {
        return "NO";
    }
}

File: Problem Solving/Algorithms/Implementation/Number Line Jumps.cs
class Solution
{
    string kangaroo(int x1, int v1, int x2, int v2)
    {
        return x2 > x1 && v1 > v2 && (x2 - x1) % (v1 - v2) == 0 ? "YES" : "NO";
    }
}
